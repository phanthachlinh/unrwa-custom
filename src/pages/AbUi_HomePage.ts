/**
* Class renders changes to the Home Page
**/
export default class AbUi_HomePage {

	constructor() {
	}

	/**
	* Changes background color and font color of the event item
	**/


	render() {
	}
}
