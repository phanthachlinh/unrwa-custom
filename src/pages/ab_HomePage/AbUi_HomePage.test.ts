import AbUi_HomePage from "./AbUi_HomePage"
import { JSDOM } from "jsdom"
declare var global: any
console.error = jest.fn((errmsg: string) => {
	throw errmsg
})
let abUiHomepageObj: any
describe('AbUi_HomePage', () => {
	describe('_changeEventItemColors', () => {
		it('should throw no event items found', () => {
			abUiHomepageObj = new AbUi_HomePage()
			expect(() => { abUiHomepageObj._changeEventItemColors() }).toThrow('No event items found')
		})
		it('should throw secondary color not found', () => {
			global.document.documentElement.innerHTML = new JSDOM(`
				<div class="event-item">
				</div>
				`).serialize()
			abUiHomepageObj = new AbUi_HomePage()
			expect(() => { abUiHomepageObj._changeEventItemColors() }).toThrow('Background (Primary color) not defined in event item no 1')
		})

		it('should throw event details not defined', () => {
			global.document.documentElement.innerHTML = new JSDOM(`
				<div class="event-item">
				<div class="event-description">
					<h3>
						<a>
						</a>
						</h3>
						</div>
					<div class="color-primary"></div>
				</div>
				`).serialize()
			abUiHomepageObj = new AbUi_HomePage()
			expect(() => { abUiHomepageObj._changeEventItemColors() }).toThrow('.eventDetails not defined in event item no 1')
		})
		it('should throw event icon not defined', () => {
			global.document.documentElement.innerHTML = new JSDOM(`
				<div class="event-item">
				<div class="event-details">
				</div>
					<div class="color-primary"></div>
				</div>
				`).serialize()
			abUiHomepageObj = new AbUi_HomePage()
			expect(() => { abUiHomepageObj._changeEventItemColors() }).toThrow('EventIcon not defined in event item no 1')
		})
		it('should match snatpshot', () => {
			global.document.documentElement.innerHTML = new JSDOM(`
				<div class="event-item">
				<div class="event-details">
				</div>
					<div class="color-primary"></div>
					<div class="event-icon">
							https://libs.iraiser.eu/users/unrwa-p2p/public/assets/collecte_unrwa_computer@2x.png
					</div>
				</div>
				`).serialize()
			abUiHomepageObj = new AbUi_HomePage()
			abUiHomepageObj._changeEventItemColors()
			expect(document.documentElement.innerHTML).toMatchSnapshot()
		})
		it('should pass', () => {
			global.document.documentElement.innerHTML = new JSDOM(`
				<div class="event-item">
						<div class="event-icon">
						</div>
						<div class="event-details">
						</div>
					<div class="color-primary"></div>
				</div>
				`).serialize()
			console.log(document.querySelector('.event-icon'))
			abUiHomepageObj = new AbUi_HomePage()
			expect(() => { abUiHomepageObj._changeEventItemColors() }).not.toThrow()
		})
	})
})
