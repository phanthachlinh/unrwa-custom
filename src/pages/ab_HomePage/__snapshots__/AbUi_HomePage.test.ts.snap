// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AbUi_HomePage _changeEventItemColors should match snatpshot 1`] = `
"<head></head><body><div class=\\"event-item\\">
				<div class=\\"event-details\\"><img class=\\"event-detail__icon\\" src=\\"
							https://libs.iraiser.eu/users/unrwa-p2p/public/assets/collecte_unrwa_computer@2x.png
					\\">
				</div>
					<div class=\\"color-primary\\"></div>
					<div class=\\"event-icon\\">
							https://libs.iraiser.eu/users/unrwa-p2p/public/assets/collecte_unrwa_computer@2x.png
					</div>
				</div>
				</body>"
`;
